ambry-cloud/src/main/java/com.github.ambry.cloud/CloudFindToken.java

```diff
@@ -160,21 +181,21 @@ public boolean equals(Object o) {
     }
     CloudFindToken that = (CloudFindToken) o;
     return version == that.version && lastUpdateTime == that.lastUpdateTime && bytesRead == that.bytesRead
-        && Objects.equals(latestBlobId, that.latestBlobId);
+        && lastUpdateTimeReadBlobIds.equals(that.lastUpdateTimeReadBlobIds);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(version, lastUpdateTime, latestBlobId, bytesRead);
+    return Objects.hash(version, lastUpdateTime, bytesRead, lastUpdateTimeReadBlobIds);
   }
 
   @Override
   public String toString() {
     StringBuilder sb = new StringBuilder();
     sb.append("version: ").append(version);
     sb.append(" lastUpdateTime: ").append(lastUpdateTime);
-    sb.append(" latestBlobId: ").append(latestBlobId);
     sb.append(" bytesRead: ").append(bytesRead);
+    sb.append(" numLastUpdateTimeBlobsRead: ").append(lastUpdateTimeReadBlobIds);
```

Comment:
Are you trying to print the set of ids or just the size of the set?  (The header doesn't match.)